manager/controlapi/service.go

```diff
@@ -90,53 +123,85 @@ func validateUpdate(uc *api.UpdateConfig) error {
 		return nil
 	}
 
+	var me MultiError
+
 	delay, err := ptypes.Duration(&uc.Delay)
 	if err != nil {
-		return err
+		me = append(me, grpc.Errorf(codes.InvalidArgument, "TaskSpec: update-delay parse error: %v", err))
+	} else {
+		if delay < 0 {
+			err = grpc.Errorf(codes.InvalidArgument, "TaskSpec: update-delay cannot be negative")
+			me = append(me, err)
+		}
 	}
 
-	if delay < 0 {
-		return grpc.Errorf(codes.InvalidArgument, "TaskSpec: update-delay cannot be negative")
+	if uc.FailureAction != api.UpdateConfig_PAUSE && uc.FailureAction != api.UpdateConfig_CONTINUE {
+		err = grpc.Errorf(codes.InvalidArgument, "TaskSpec: update-on-failure should be in range 0-1")
```

Comment:
I'd prefer not to mention protobuf field numbers in error messages. They won't be meaningful to the user.